### ISSUE TITLE
Four-digit expressions separated by a hyphen misidentified as timestamp

### ISSUE THREAD

**Issue by ghost** [wanasit/chrono#432]
For example, the string `2012-1400` is parsed as:

 ```
{"end": null, "index": 0, "refDate": 2022-02-09T17:29:15.341Z, "reference": {"instant": 2022-02-09T17:29:15.341Z, "timezoneOffset": -0}, "start": {"impliedValues": {"day": 9, "millisecond": 0, "month": 2, "second": 0, "year": 2022}, "knownValues": {"hour": 20, "meridiem": 1, "minute": 12, "timezoneOffset": -840}, "reference": {"instant": 2022-02-09T17:29:15.341Z, "timezoneOffset": -0}}, "text": "2012-1400"}
```

I think this is due to the first four-digit segment being identified as a 24h time (20:12) and the second four-digit segment being identified as a timezone offset (UTC-1400?  although the parsed output shows a timezone offset of -840..)

**Comment by JoakimNil** [comment]
I too experience this issue. It seems closely related to your other issue, #420 . It looks like an attempt was made to filter matches like these in 053cc8f92d8c2e63c45e4720597608afd3631f21 (see `checkAndReturnWithFollowingPattern` in `AbstractTimeExpressionParser`), but the function fails to filter many cases. I was thinking about submitting a PR, but I'm not sure if matches like these are intentional, and if not, how strict the parsing should be - can you comment on that @wanasit? Should stuff like the above post, the example from #420, "2.25-2", "at 10-25" and "10-25" be filtered?

In the meantime, a workaround is to add a custom refiner to the chrono instance that filters out matches like this. E.g.:

    chrono.refiners.push({refine: (_context, results: Array<ParsingResult>) => results.filter(res => !new RegExp(String.raw`^[\d\s-.:]+$`).test(res.text))});

It's not a perfect solution: in particular, it will fail to filter out results like "at 10-25", but it removes most false matches for me.

**Comment by wanasit** [comment]
@JoakimNil Thanks for letting me know about the issue. The `AbstractTimeExpressionParser` causes the problem as you mentioned. The `checkAndReturnWithFollowingPattern` also has some issues.

I addressed that in [4756d0d]. (https://github.com/wanasit/chrono/commit/4756d0dd5b816d3e5aa3b90024e064750ce82f98). 

> I was thinking about submitting a PR, but I'm not sure if matches like these are intentional, and if not, how strict the parsing should be
FYI. The tests are pretty extensive. If your change doesn't break any test, you should be fine.
